### PR TITLE
MRC-1631 - Directions Beta, add step by step support to the multi-floor demo

### PR DESCRIPTION
**Pull request by vercasson**
- [x] This change is documented in CHANGELOG.md

### CODE REVIEW

File: demo/directions-overview-multifloor/index.html [hunk 1]
```diff
@@ -4,154 +4,167 @@
...
+            stroke: "yellow",
+            strokeOpacity: ".5",
             strokeWidth: 8,
             strokeLineJoin: "round",
             strokeLineCap: "round",
+            // markerStart: "url(#arrow)",
+            // markerMid: "url(#arrow)",
+            // markerEnd: "url(#arrow)",
+          },
+          // {
```
Comment (tarakreft) [inline demo/directions-overview-multifloor/index.html:148]: Any reason you are keeping the commented code?

File: demo/directions-overview-multifloor/index.html [hunk 2]
```diff
@@ -161,60 +174,375 @@ <h3>Directions Request</h3>
...
-              overlays: [overlay],
-            };
-          } else {
-            return {
+          },
+          // {
+          //   type: "path",
+          //   id: "mypath",
+          //   shape: stepShape,
+          //   stroke: "blue",
```
Comment (tarakreft) [inline demo/directions-overview-multifloor/index.html:182]: same here, more commented code

File: src/OverlayMarker.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ const OverlayMarker: FunctionComponent<OverlayMarkerProps> = ({
       fill={fill}
       fill-opacity={fillOpacity}
       stroke={stroke}
-      stroke-width={strokeWidth / mapZoomFactor}
```
Comment (tarakreft) [inline src/OverlayMarker.tsx:76]: why doesn't this need the mapZoomFactor anymore?

File: demo/directions-overview-multifloor/index.html [hunk 4]
```diff
@@ -4,154 +4,167 @@
...
-            let map_id = current.map_id;
-            let points = current.points;
-
-            if (floor !== map_id) {
-              floor_index++;
-              accumulator[floor_index] = {
-                floorID: map_id,
-                steps: [],
-              };
+      function stepsToPolygonPoints(steps, mapID) {
```
Comment (tarakreft) [inline demo/directions-overview-multifloor/index.html:83]: I'm curious why all these functions (stepsToPolygonPoints, generateRouteOverlays, etc) are defined in the demo. I would expect there to be functions that a use can call in the webSdk and then the demo shows users how to leverage the functions to build their overlays.

**Review by tarakreft — APPROVED** [review]

File: demo/directions-overview-multifloor/index.html [hunk 5]
```diff
@@ -4,154 +4,167 @@
...
             type: "polyline",
-            points,
-            stroke: "orange",
+            points: polylinePoints,
+            stroke: "yellow",
+            strokeOpacity: ".5",
             strokeWidth: 8,
             strokeLineJoin: "round",
             strokeLineCap: "round",
+            // markerStart: "url(#arrow)",
```
Comment (amyhliao) [inline demo/directions-overview-multifloor/index.html:144]: LGTM, are these comments needed?

File: demo/directions-overview-multifloor/index.html [hunk 6]
```diff
@@ -161,60 +174,375 @@ <h3>Directions Request</h3>
...
             },
-          };
-          if (floorID === currentFloorID) {
-            return {
-              overlays: [overlay],
-            };
-          } else {
-            return {
+          },
+          // {
```
Comment (amyhliao) [inline demo/directions-overview-multifloor/index.html:178]: These too.

**Review by amyhliao — APPROVED** [review]

File: src/OverlayMarker.tsx (hunk 3 above)
Comment (vercasson) [reply src/OverlayMarker.tsx:76]: markers don't need this

File: demo/directions-overview-multifloor/index.html (hunk 4 above)
Comment (vercasson) [reply demo/directions-overview-multifloor/index.html:83]: These will most likely be checked in at one point. Not ready for that yet.

File: demo/directions-overview-multifloor/index.html (hunk 5 above)
Comment (vercasson) [reply demo/directions-overview-multifloor/index.html:144]: nope

File: demo/directions-overview-multifloor/index.html (hunk 1 above)
Comment (vercasson) [reply demo/directions-overview-multifloor/index.html:148]: Thanks, I'll delete these

Comment (vercasson) [reply demo/directions-overview-multifloor/index.html]: None, will remove

File: demo/directions-overview-multifloor/index.html (hunk 2 above)
Comment (vercasson) [reply demo/directions-overview-multifloor/index.html]: Thanks, will remove

**Pull request merged**